File: 8-kyu/total-amount-of-points/c++/solution.cpp
#include <array>
#include <string>

int points(const std::array<std::string, 10>& games) {
    int ret = 0;
    for (const std::string &score : games) {
        auto colon_pos = score.find(':');
        auto x_score_str = score.substr(0, colon_pos);
        auto y_score_str = score.substr(colon_pos + 1);
        auto x_score = std::stoi(x_score_str);
        auto y_score = std::stoi(y_score_str);
        ret += (x_score > y_score ? 3 : x_score == y_score ? 1 : 0);
    }
    return ret;
}

File: 8-kyu/total-amount-of-points/javascript/solution.js
const points = games => {
  let ret = 0;
  for (const score of games) {
    const split = score.split(":");
    if (split[0] > split[1]) {
      ret += 3;
    } else if (split[0] === split[1]) {
      ret++;
    }
  }
  return ret;
};
